lib/ansible/modules/cloud/amazon/iam_user_info.py

```diff
@@ -0,0 +1,184 @@
+version_added: "2.10"
+author:
+  - Constantin Bugneac (@Constantin07)
+  - Abhijeet Kasurde (@Akasurde)
+options:
+  name:
+    description:
+     - The name of the IAM user to look for.
+    required: false
+    type: str
+  group:
+    description:
+     - The group name name of the IAM user to look for. Mutually exclusive with C(path).
+    required: false
+    type: str
+  path:
+    description:
+     - The path to the IAM user. Mutually exclusive with C(group).
+     - If specified, then would get all user names whose path starts with user provided value.
+    required: false
+    default: '/'
+    type: str
+requirements:
+  - botocore
+  - boto3
+extends_documentation_fragment:
+  - aws
+  - ec2
+'''
+
+EXAMPLES = r'''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+# Gather facts about "test" user.
+- name: Get IAM user facts
+  iam_user_info:
+    name: "test"
+
+# Gather facts about all users in the "dev" group.
+- name: Get IAM user facts
+  iam_user_info:
+    group: "dev"
+
+# Gather facts about all users with "/division_abc/subdivision_xyz/" path.
+- name: Get IAM user facts
+  iam_user_info:
+    path: "/division_abc/subdivision_xyz/"
+'''
+
+RETURN = r'''
+iam_users:
+    description: list of maching iam users
+    returned: success
+    type: complex
+    contains:
+        arn:
+            description: the ARN of the user
+            returned: if user exists
+            type: str
+            sample: "arn:aws:iam::156360693172:user/dev/test_user"
+        create_date:
+            description: the datetime user was created
+            returned: if user exists
+            type: str
+            sample: "2016-05-24T12:24:59+00:00"
+        password_last_used:
+            description: the last datetime the password was used by user
+            returned: if password was used at least once
+            type: str
+            sample: "2016-05-25T13:39:11+00:00"
+        path:
+            description: the path to user
+            returned: if user exists
+            type: str
+            sample: "/dev/"
+        user_id:
+            description: the unique user id
+            returned: if user exists
+            type: str
+            sample: "AIDUIOOCQKTUGI6QJLGH2"
+        user_name:
+            description: the user name
+            returned: if user exists
+            type: str
+            sample: "test_user"
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import boto3_conn, camel_dict_to_snake_dict, ec2_argument_spec, get_aws_connection_info
+
+try:
+    import botocore
+    from botocore.exceptions import ClientError, ParamValidationError
+except ImportError:
+    pass  # caught by AnsibleAWSModule
+
+
+def list_iam_users(connection, module):
+
+    name = module.params.get('name')
+    group = module.params.get('group')
+    path = module.params.get('path')
+
+    params = dict(MaxItems=1000)
```

Comment:
This could in theory catch people out.  Unfortunately I don't have any experience with boto pagination...

Reply:
https://github.com/ansible/ansible/pull/23382/commits/83e179a77b828b6e2fa3e472d74ad0fdae2e5197 updates this.